include/realizations/config/layer.hpp

```diff
@@ -0,0 +1,103 @@
+#ifndef NGEN_REALIZATION_CONFIG_LAYER_H
+#define NGEN_REALIZATION_CONFIG_LAYER_H
+
+#include <boost/property_tree/ptree.hpp>
+
+#include "LayerData.hpp"
+#include "config.hpp"
+
+namespace realization{
+  namespace config{
+
+  /**
+   * @brief Layer configuration information
+   * 
+   */
+  struct Layer{
+    /**
+     * @brief Construct a new default surface layer
+     * 
+     * Default layers are surface layers (0) with 3600 second time steps
+     * 
+     */
+    Layer():descriptor( {"surface layer", "s", 0, 3600 } ){};
+
+    /**
+     * @brief Construct a new Layer from a property tree
+     * 
+     * @param tree 
+     */
+    Layer(const boost::property_tree::ptree& tree):formulation(tree){
+        auto name = tree.get_optional<std::string>("name");
+        if(!name) missing_keys.push_back("name");
+        auto unit = tree.get<std::string>("time_step_units", "s");
+
+        auto id = tree.get_optional<int>("id");
+        if(!id) missing_keys.push_back("id");
+
+        auto ts = tree.get_optional<double>("time_step");
+        //TODO is time_step required? e.g. need to add to missing_keys?
+        auto tmp = tree.get_optional<std::string>("domain");
+        if(tmp) domain = *tmp;
+        if(missing_keys.empty()){
+            descriptor = {*name, unit, *id, *ts};// ngen::LayerDescription(  );
+        }
+        else{
+            //FIXME then what?
```

Comment:
At the very least, for now, should throw an exception here so that it doesn't get ignored.

Reply:
Disregard, missed the TODO in the PR description

Reply:
refactored in 88f3d23de09130996bcb602acb0a0aca27b0e88b